pkg/wireguard/agent/agent.go

```diff
@@ -450,6 +452,50 @@ func (a *Agent) OnIPIdentityCacheGC() {
 	// ignored
 }
 
+// Status returns the state of the Wireguard tunnel managed by this instance.
+// If withPeers is true, then the details about each connected peer are
+// are populated as well.
+func (a *Agent) Status(withPeers bool) (*models.WireguardStatus, error) {
+	a.Lock()
+	dev, err := a.wgClient.Device(types.IfaceName)
+	a.Unlock()
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get device: %w", err)
+	}
+
+	var peers []*models.WireguardPeer
+	if withPeers {
+		peers = make([]*models.WireguardPeer, 0, len(dev.Peers))
+		for _, p := range dev.Peers {
+			allowedIPs := make([]string, 0, len(p.AllowedIPs))
+			for _, ip := range p.AllowedIPs {
+				allowedIPs = append(allowedIPs, ip.String())
+			}
+
+			peer := &models.WireguardPeer{
+				PublicKey:         p.PublicKey.String(),
+				Endpoint:          p.Endpoint.String(),
+				LastHandshakeTime: strfmt.DateTime(p.LastHandshakeTime),
+				AllowedIps:        allowedIPs,
+			}
+			peers = append(peers, peer)
```

Comment:
nit: in both cases, the slice size is known beforehand. Therefore, you could use the loop index instead of appending to the slice.
```suggestion
		peers = make([]*models.WireguardPeer, len(dev.Peers))
		for i, p := range dev.Peers {
			allowedIPs := make([]string, len(p.AllowedIPs))
			for j, ip := range p.AllowedIPs {
				allowedIPs[j] = ip.String()
			}

			peers[i] = &models.WireguardPeer{
				PublicKey:         p.PublicKey.String(),
				Endpoint:          p.Endpoint.String(),
				LastHandshakeTime: strfmt.DateTime(p.LastHandshakeTime),
				AllowedIps:        allowedIPs,
			}
```

Reply:
Thanks for the review! I personally prefer `append` instead of assignment, as it is less error prone to future modifications such as early returns/filtering etc.